File: apps/api/app/schemas/risk_alerts.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import FixtureEnvelope

RiskCategory = Literal["정책", "출처", "시장", "승인", "데이터"]
RiskSeverity = Literal["중대", "높음", "보통", "낮음"]
RiskEventLinkPage = Literal["data", "policy", "trades"]


class RiskEvent(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    occurredAt: str
    title: str
    decisionRef: str
    category: RiskCategory
    severity: RiskSeverity
    status: str
    summary: str
    cause: str
    action: str
    policy: str
    linkPage: RiskEventLinkPage | None
    linkText: str


class RiskAlertsData(BaseModel):
    model_config = ConfigDict(extra="forbid")

    events: list[RiskEvent] = Field(min_length=1)
    safetyCopy: str


class RiskAlertsEnvelope(FixtureEnvelope):
    data: RiskAlertsData

File: apps/web/src/types/riskAlerts.ts
import type { components } from "./api.generated";
import type { Api, FixtureEnvelope } from "./common";

export type RiskCategory = "정책" | "출처" | "시장" | "승인" | "데이터";
export type RiskSeverity = "중대" | "높음" | "보통" | "낮음";

export type RiskEvent = Api<components["schemas"]["RiskEvent"]>;
export type RiskAlertsData = Api<components["schemas"]["RiskAlertsData"]>;
export type RiskAlertsEnvelope = FixtureEnvelope<RiskAlertsData>;
